src/index.tsx

```diff
@@ -4,17 +4,33 @@ import 'react-app-polyfill/ie11';
 import * as React from 'react';
 import * as ReactDOM from 'react-dom';
 import { BrowserRouter } from 'react-router-dom';
-import { I18nextProvider } from 'react-i18next';
-import { getInstance as getI18nInstance } from './i18n';
-import App from './pages/App';
+import * as i18n from './i18n';
+import { ErrorBoundary } from './errorBoundary';
+import Shell from './shell/shell';
 import * as serviceWorker from './serviceWorker';
 
+// initialize the i18next instance before rendering anything:
+i18n.initInstance();
+
+// Render <Shell> in the #root dom element. Shell requires:
+// 1. <BrowserRouter> that provides the routing functionality.
+// 2. <Suspense> that provides a fallback when we're async-loading the js bundles
+//    and i18n resjsons. The fallback should result in an empty #root element,
+//    which triggers the default loading styles in public/index.html.
+// 3. <ErrorBoundary> that displays an error message if something fails with the 
+//    async request. At this point, we can't rely on anything else working so 
+//    just display a static hardcoded message.
+// 4. <StrictMode> to highlight problems with the application code.
 ReactDOM.render(
-    <I18nextProvider i18n={getI18nInstance()}>
-        <BrowserRouter>
-            <App />
-        </BrowserRouter>
-    </I18nextProvider>,
+    <React.StrictMode>
+        <ErrorBoundary message='Something went wrong'>
```

Comment:
nit: can the message be localized

Reply:
Nope, as the comment mentions, this is a last chance error handler. If the error has bubbled up all the way here, we can't rely on anything else (including i18next) working, so we need to default to a hardcoded message.